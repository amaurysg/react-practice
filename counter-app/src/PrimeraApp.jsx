import React, { useState } from 'react'
import PropTypes from 'prop-types'

const PrimeraApp = ({saludo, subtitulo}) => {

  //Props
 console.log("props, desestructurado:",saludo)
/*   const [user, setuser] = useState(false) */
const user = true
const names = ['Amaury',9,8]
const saludos = {
  names:'amaury', 
  apellido:'soto',
  edad:25
}

  return (
        
       <>

            <h1>{saludo} !</h1>
            <p>{subtitulo}</p>

         {/*    {
              !user ? 
              <h1>No hay User</h1> 
              :  
              <div>
              
                <h1>Welcome {JSON.stringify(saludos,null, 3)} </h1>
                <h1>Welcome {saludos.names} </h1>
                <h1> {saludo} </h1>
                <h2> {subtitulo} </h2>
             
                <form action="">
                  <input type="text"/>
                  <input type="text"/>
                </form>
              </div>
            } */}
         

       </> 
       
    
  )
}
//PropTypes //
PrimeraApp.propTypes={
  saludo: PropTypes.string.isRequired
}

//DefaultProps//
PrimeraApp.defaultProps = {
  subtitulo : 'Soy subtitulo (default)'
}


export default PrimeraApp
