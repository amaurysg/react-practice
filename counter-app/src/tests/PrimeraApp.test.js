import React from 'react'
import { render } from "@testing-library/react"
import PrimeraApp from "../PrimeraApp"
import Enzyme from 'enzyme';
import { shallow } from 'enzyme'

describe('Pruebas Comp <PrimeraApp/>', () => {
  
/*   test('Debe mostrar msj Hola Amaury ', () => {

      //We need render the component 
      //This message to evalued
      const saludo = 'Hola Amaury'

      //This we can use method render from  testing-library //
      const {getByText}= render(<PrimeraApp saludo={saludo} />)
    


     
     

      
      //Method toBeInTheDocument from import setupTests.js
      //Here only test about if exist saludo in this component
      expect( getByText(saludo) ).toBeInTheDocument()


  }) */


  test('PrimeraApp correctamente <PrimeraApp/> ', () => {

    const saludo = 'Hola Amaury'
    const wrapper = shallow( <PrimeraApp saludo={saludo} />)

    //We expect that the component render ok 
    expect( wrapper).toMatchSnapshot()
    
  })
  test('Debe mostrar subtitulo enviado por props ', () => {

    const saludo = "Hola, Amaury"
    const subTitulo= "Soy un subtitulo"
    const wrapper = shallow(
       <PrimeraApp
        saludo={saludo}
        subtitulo={subTitulo}
        />)
    
        const textParrafo = wrapper.find('p').text()
        console.log(textParrafo)

        expect(textParrafo).toBe(subTitulo)
  })

  
  

})
