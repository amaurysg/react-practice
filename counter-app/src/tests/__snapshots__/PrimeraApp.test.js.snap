// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Comp <PrimeraApp/> PrimeraApp correctamente <PrimeraApp/>  1`] = `
<Fragment>
  <h1>
    Hola Amaury
     !
  </h1>
  <p>
    Soy subtitulo (default)
  </p>
</Fragment>
`;
